sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/JsonToRow.java

```diff
@@ -116,4 +131,267 @@ private ObjectMapper objectMapper() {
       return this.objectMapper;
     }
   }
+
+  /**
+   * Enable Dead letter support. If this value is set errors in the parsing layer are returned as
+   * Row objects within a {@link ParseResult}
+   *
+   * <p>You can access the results by using:
+   *
+   * <p>ParseResult results = jsonPersons.apply(JsonToRow.withDeadLetter(PERSON_SCHEMA));
+   *
+   * <p>{@link ParseResult#getResults()}
+   *
+   * <p>{@Code PCollection<Row> personRows = results.getResults()}
+   *
+   * <p>{@link ParseResult#getFailedToParseLines()}
+   *
+   * <p>{@Code PCollection<Row> errorsLines = results.getFailedToParseLines()}
+   *
+   * <p>To access the reason for the failure you will need to first enable extended error reporting.
+   * {@Code ParseResult results =
+   * jsonPersons.apply(JsonToRow.withDeadLetter(PERSON_SCHEMA).withExtendedErrorInfo()); }
+   *
+   * <p>{@link ParseResult#getFailedToParseLinesWithErr()}
+   *
+   * <p>{@Code PCollection<Row> errorsLinesWithErrMsg = results.getFailedToParseLines()}
+   *
+   * @return {@link JsonToRowWithErrFn}
+   */
+  @Experimental(Kind.SCHEMAS)
+  public static JsonToRowWithErrFn withDeadLetter(Schema rowSchema) {
```

Comment:
I think we should name this something different, as not everyone knows the term dead letter.

Reply:
Changed to withExceptionReporting.